code-challenges.js

```diff
@@ -28,3 +60,85 @@ var hand2 = [5, 5, 3, 3, 4]
+    // NOTE: the above code looks terrible in ternary because of the chaining of methods so I avoided using ternary
+}
+
+console.log(fullHouseCheck(hand1))
+console.log(fullHouseCheck(hand2))
+console.log(fullHouseCheck(hand3))
+
+
+// MY SECOND ATTEMPT USING DIFFERENT METHODS
+// I originally just wanted to filter for a triple and a pair separately assigning booleans to separate variables
+// if triple was true and if pair was true, then the array contains a full house
+// otherwise if either variables are false, full house returns false
+// I had such a hard time trying to do this elegantly (I had to hard code index values originally to count duplicates)
+// But I furthered by research! (I didn't want to give up too soon even after seeing Brian's solution)
+// I found a way to build an object! (a lot of work reading through docs and stackoverflow but
+// it was a great learning experience)
+// I also learned how to access the key by using the Object.keys(nameOfObject) method! (it's great btw)
+// ALSO! counts[value] is the same as counts.values BUT for some reason in the forEach H-O-F, it doesn't
+// recognize .value as the value parameter for forEach
+// For this reason, I had to use bracket notation instead of dot notation... Idk why but I want to know why (haha)
+// Is it because when using bracket notation, the property name/key has to be a String?
+const fullHouse2 = (array) => {
+    // create an empty array that we will eventually fill with the occurence of each number in the given array
+    let counts = {}
+    // value represents the number in the user's array (i.e. hand1, hand2, hand3)
+    // use the forEach method to go through each value which will be the key in the object counts
+    // count each occurence of the value by incrementing its count everytime it appears in the forEach loop
+    // OR if the value appears for the first time, assign the key's value to 0 before incrementing
+    array.forEach(value => counts[value] = (counts[value] || 0) + 1)
+    // console.log(counts)
+    // filter for keys that have a 3 as a value which are the hand's triple
+    let triple = Object.keys(counts).filter(key => counts[key] === 3).length === 1
+    // console.log(triple)
+    // filter for keys that have 2 as a value which are the hand's pair
+    let pair = Object.keys(counts).filter(key => counts[key] === 2).length === 1
+    // console.log(pair)
+    // Empty arrays [] are considered truthy so I had to fix the if-else statements--was getting undefined too
+    // Boolean(triple) === true && Boolean(pair) === true ? true : false
+    // I just decided to use .length instead because if the triple and single pair are true, there must only be a single value in the array
+    // this will check for double pairs which we don't want and gives a falsey
+    // did away with if and else statement because JS will do the evalution of true/false values
+    // true && true => true
+    // false && false => false
+    return triple && pair
+}
```

Comment:
I know you felt really stuck on this one but you came out the other side, girl! both of these solutions are awesome and I commend your tenacity to do a second attempt all on your own. GREAT work!